oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/ClusterNodeInfo.java

```diff
@@ -615,6 +650,18 @@ private static Map.Entry<ClusterNodeInfo, Long> createInstance(DocumentStore sto
             // if we get here the cluster node entry is inactive. if recovery
             // was needed, then it was successful
 
+            if (reuseAfterRecoveryMillis > 0) {
+                Long lastRecoveryTime = (Long) doc.get(RECOVERY_TIME_KEY);
```

Comment:
We are setting `RECOVERY_TIME_KEY` as null in many places and this could lead to an NPE in case it is null here.

I would suggest using default value in place of null when we are setting `RECOVERY_TIME_KEY` or doing a null check here before casting to `Long`.

Reply:
If the document doesn't contain a `"recoveryTime"` key or the value of that key is `null`, then this will result in `lastRecoveryTime` here being null. No NPE is thrown. 

The intent was:
* `"recoveryTime"` is null (non existent) in legacy cases : in this case no delay should apply (hence it skips the blow below)
* `"recoveryTime"` is null (non existent) after a graceful shutdown : in this cases also no delay should apply
* `"recoveryTime"` is set to an actual value at the end of a recovery : only in this case should the blow below be handled

All of these cases should be handled by the null check in the next line, or am I missing something?